crates/kernel/src/tasks/scheduler.rs

```diff
@@ -73,21 +72,63 @@ pub struct Scheduler {
     running: Arc<AtomicBool>,
     database: Arc<dyn Database + Send + Sync>,
     next_task_id: AtomicUsize,
-    inner: RwLock<Inner>,
+
+    /// The internal task queue which holds our suspended tasks, and control records for actively
+    /// running tasks.
+    task_q: Arc<Mutex<TaskQ>>,
```

Comment:
Is this the part where we're not faffing about with concurrent hashmaps? (Do you expect this to ever be a bottleneck?)

Reply:
Yeah, the earlier version of this kind of thing was using dashmap, and there were deadlocks.

There are probably ways to optimize here to avoid bottlenecking -- which really is just about lots of tasks spawning/completing concurrently, but that's a problem I'd like to have in the future.